From ca9d9eb55ed6300ef11c8d78e185d55f38340ed3 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Thu, 23 Oct 2014 11:30:14 +0200
Subject: [LIBREPORT PATCH] ureport: fall back to the hardcoded rhsm cert dir

This commit adds a fallback execution path for cases where getting the
rhsm entitlement dir path via an execution of python from the
librepor-web library fails for any reason.

See commit 2b20c9f91342da7744ae40ee623735ab95f83219
Related #1140224

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/lib/ureport.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/lib/ureport.c b/src/lib/ureport.c
index d0bf50b..76be572 100644
--- a/src/lib/ureport.c
+++ b/src/lib/ureport.c
@@ -32,6 +32,7 @@
 
 #define RHSM_WEB_SERVICE_URL "https://api.access.redhat.com/rs/telemetry/abrt"
 
+#define RHSMENT_PEM_DIR_PATH "/etc/pki/entitlement"
 #define RHSMENT_ENT_DATA_BEGIN_TAG "-----BEGIN ENTITLEMENT DATA-----"
 #define RHSMENT_ENT_DATA_END_TAG "-----END ENTITLEMENT DATA-----"
 #define RHSMENT_SIG_DATA_BEGIN_TAG "-----BEGIN RSA SIGNATURE-----"
@@ -92,7 +93,7 @@ rhsm_config_get_entitlement_cert_dir(void)
 error:
     error_msg("Failed to get 'rhsm':'entitlementCertDir' from rhsm.config python module.");
     free(result);
-    return NULL;
+    return xstrdup(RHSMENT_PEM_DIR_PATH);
 }
 
 void
-- 
1.8.3.1

